flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java

```diff
@@ -293,6 +293,61 @@ public void run() {
 		testHarness.close();
 	}
 
+	/**
+	 * This test is meant to assure that testAtLeastOnceProducer is valid by testing that if flushing is disabled,
+	 * the snapshot method does indeed finishes without waiting for pending records;
+	 * we set a timeout because the test will not finish if the logic is broken
+	 */
+	@Test(timeout=5000)
+	public void testDoesNotWaitForPendingRecordsIfFlushingDisabled() throws Throwable {
+		final OneShotLatch inputLatch = new OneShotLatch();
+
+		final DummyFlinkKafkaProducer<String> producer = new DummyFlinkKafkaProducer<>(
```

Comment:
That makes sense. I was just reusing the {{MockProducer}} implementation in this class, which is needed in other tests to be able to provide success / failure fake callback completions. In this test we don't need to do that, so a simple {{mock(KafkaProducer.class)}} should do.